src/tasks/task-two-frames-equality.cpp

```diff
@@ -0,0 +1,178 @@
+using namespace math;
+using namespace trajectories;
+using namespace pinocchio;
+
+TaskTwoFramesEquality::TaskTwoFramesEquality(const std::string& name,
+                                             RobotWrapper& robot,
+                                             const std::string& frameName1,
+                                             const std::string& frameName2)
+    : TaskMotion(name, robot),
+      m_frame_name1(frameName1),
+      m_frame_name2(frameName2),
+      m_constraint(name, 6, robot.nv()) {
+  assert(m_robot.model().existFrame(frameName1));
+  assert(m_robot.model().existFrame(frameName2));
+  m_frame_id1 = m_robot.model().getFrameId(frameName1);
+  m_frame_id2 = m_robot.model().getFrameId(frameName2);
+
+  m_v_ref.setZero();
+  m_a_ref.setZero();
+  m_wMl1.setIdentity();
+  m_wMl2.setIdentity();
+  m_p_error_vec.setZero(6);
+  m_v_error_vec.setZero(6);
+  m_p.resize(12);
+  m_v.resize(6);
+  m_p_ref.resize(12);
+  m_v_ref_vec.resize(6);
+  m_Kp.setZero(6);
+  m_Kd.setZero(6);
+  m_a_des.setZero(6);
+  m_J1.setZero(6, robot.nv());
+  m_J2.setZero(6, robot.nv());
+  m_J1_rotated.setZero(6, robot.nv());
+  m_J2_rotated.setZero(6, robot.nv());
+
+  m_mask.resize(6);
+  m_mask.fill(1.);
+  setMask(m_mask);
+}
+
+void TaskTwoFramesEquality::setMask(math::ConstRefVector mask) {
+  TaskMotion::setMask(mask);
+  int n = dim();
+  m_constraint.resize(n, (unsigned int)m_J1.cols());
+  m_p_error_masked_vec.resize(n);
+  m_v_error_masked_vec.resize(n);
+  m_drift_masked.resize(n);
+  m_a_des_masked.resize(n);
+}
+
+int TaskTwoFramesEquality::dim() const { return (int)m_mask.sum(); }
+
+const Vector& TaskTwoFramesEquality::Kp() const { return m_Kp; }
+
+const Vector& TaskTwoFramesEquality::Kd() const { return m_Kd; }
+
+void TaskTwoFramesEquality::Kp(ConstRefVector Kp) {
+  assert(Kp.size() == 6);
+  m_Kp = Kp;
+}
+
+void TaskTwoFramesEquality::Kd(ConstRefVector Kd) {
+  assert(Kd.size() == 6);
+  m_Kd = Kd;
+}
+
+const Vector& TaskTwoFramesEquality::position_error() const {
+  return m_p_error_masked_vec;
+}
+
+const Vector& TaskTwoFramesEquality::velocity_error() const {
+  return m_v_error_masked_vec;
+}
+
+const Vector& TaskTwoFramesEquality::getDesiredAcceleration() const {
+  return m_a_des_masked;
+}
+
+Vector TaskTwoFramesEquality::getAcceleration(ConstRefVector dv) const {
+  return m_constraint.matrix() * dv + m_drift_masked;
+}
+
+Index TaskTwoFramesEquality::frame_id1() const { return m_frame_id1; }
+
+Index TaskTwoFramesEquality::frame_id2() const { return m_frame_id2; }
+
+const ConstraintBase& TaskTwoFramesEquality::getConstraint() const {
+  return m_constraint;
+}
+
+const ConstraintBase& TaskTwoFramesEquality::compute(const double,
+                                                     ConstRefVector,
+                                                     ConstRefVector,
+                                                     Data& data) {
+  // Calculating task with formulation: [J1 - J2   0   0] y = [-J1dot*v +
```

Comment:
This comment is misleading because it suggests that the constraint matrix to be computed is the one that's gonna be multiplied by the vector y containing all the decision variables. Instead, this method has to compute the matrix that multiplies just the robot accelerations `dv`.